src/helpers/redirects.ts

```diff
@@ -102,6 +109,36 @@ export const generateRedirects = async ({
     ...(await getPreviewRewrites({ basePath, appDir })),
   )
 
+  const middleware = await getMiddleware(netlifyConfig.build.publish)
+  const routesThatMatchMiddleware = new Set<string>()
+
+  const handlerRewrite = (from: string) => ({
+    from: `${basePath}${from}`,
+    to: HANDLER_FUNCTION_PATH,
+    status: 200,
+  })
+
+  // Routes that match middleware need to always use the SSR function
+  // This generates a rewrite for every middleware in every locale, both with and without a splat
+  netlifyConfig.redirects.push(
+    ...middleware
+      .map((route) => {
+        const unlocalized = [handlerRewrite(`${route}`), handlerRewrite(`${route}/*`)]
+        if (i18n?.locales?.length > 0) {
+          const localized = i18n?.locales?.map((locale) => [
```

Comment:
```suggestion
          const localized = i18n.locales.map((locale) => [
```
Since the above check prevents it, you don't need to have the optional check here